pkg/build/cleanup.go

```diff
@@ -11,8 +11,16 @@ import (
 // temporary directories created by STI build and it also cleans the temporary
 // Docker images produced by LayeredBuild
 type DefaultCleaner struct {
-	util.FileSystem
-	docker.Docker
+	fs     util.FileSystem
+	docker docker.Docker
```

Comment:
actually i can even think of a reason you would not want to embed the mutex in this way:  you've now exposed the mutex to any code w/ a reference to your Counter, which means it could potentially lock or unlock the mutex in unsafe ways.  violates data encapsulation.